internal/test/smoke/suite_test.go

```diff
@@ -0,0 +1,412 @@
+	if resp, err = s.DoRequest(
+		http.MethodPost,
+		url,
+		headers,
+		&public.DomainRegTokenRequest{
+			DomainType: "rhel-idm",
+		},
+	); err != nil {
+		return nil, fmt.Errorf("failure when POST %s: %w", url, err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failure when POST %s: expected '%d' but got '%d'", url, http.StatusOK, resp.StatusCode)
+	}
+	var data []byte
+	data, err = io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failure when reading body for POST %s because an empty response", url)
+	}
+	var token *public.DomainRegTokenResponse = &public.DomainRegTokenResponse{}
+	err = json.Unmarshal(data, token)
+	if err != nil {
+		return nil, fmt.Errorf("failure when unmarshalling the information for POST %s", url)
+	}
+	return token, nil
+}
+
+// RegisterIpaDomain is a helper function to register a domain with the API
+// for a rhel-idm domain using the OrgID assigned to the unit test.
+// Return the token response or error.
+func (s *SuiteBase) RegisterIpaDomain(domain *public.Domain) (*public.Domain, error) {
+	var headers http.Header = http.Header{}
+	var resp *http.Response
+	var err error
+	var token *public.DomainRegTokenResponse = nil
+
+	if token, err = s.CreateToken(); err != nil {
+		s.FailNow("Error creating a token for registering a rhel-idm domain", "%s", err.Error())
+	}
+
+	url := s.DefaultPublicBaseURL() + "/domains"
+	headers.Add("X-Rh-Idm-Registration-Token", token.DomainToken)
+	headers.Add("X-Rh-Insights-Request-Id", "get_token")
+	headers.Add("X-Rh-Identity", header.EncodeXRHID(&s.SystemXRHID))
+	headers.Add("X-Rh-Idm-Version", header.EncodeXRHIDMVersion(&header.XRHIDMVersion{
+		IPAHCCVersion:      "1.0.0",
+		IPAVersion:         "4.19.0",
+		OSReleaseID:        "9.3",
+		OSReleaseVersionID: "redhat-9.3",
+	}))
+	if resp, err = s.DoRequest(
+		http.MethodPost,
+		url,
+		headers,
+		domain,
+	); err != nil {
+		return nil, fmt.Errorf("failure when %s %s: %w", http.MethodPost, url, err)
+	}
+	// FIXME This should be http.StatusCreated instead of StatusOK
+	if resp.StatusCode != http.StatusCreated {
+		return nil, fmt.Errorf("failure when POST %s: expected '%d' but got '%d'", url, http.StatusOK, resp.StatusCode)
+	}
+	var data []byte
+	data, err = io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failure when reading body for POST %s because an empty response", url)
+	}
+	var createdDomain *public.Domain = &public.Domain{}
+	err = json.Unmarshal(data, createdDomain)
+	if err != nil {
+		return nil, fmt.Errorf("failure when unmarshalling the information for %s %s", http.MethodPost, url)
+	}
+	return createdDomain, nil
+}
+
+// RunTestCase run test for one specific testcase
+func (s *SuiteBase) RunTestCase(testCase *TestCase) {
+	t := s.T()
+
+	// GIVEN
+	var (
+		body []byte
+		resp *http.Response
+		err  error
+	)
+
+	// WHEN
+	resp, err = s.DoRequest(testCase.Given.Method, testCase.Given.URL, testCase.Given.Header, testCase.Given.Body)
```

Comment:
It is not checked that for the request only `Body`or `BodyBytes` are used; I suggest to add the check and fail with informative error how to solve for the developer in the future.